Add Case Studies Infrastructure

hschafer opened the pull request:
Added the notion of a "case study" to the table of contents and added prev/next buttons to navigate.

Also a slight CSS upgrade of the header to make the experience cleaner.

hschafer commented:
Sometimes I want to avoid work so I write useless frontend code to avoid responsibility 😤

Rockett8855 commented on website/components/layout.js:
```diff
@@ -8,14 +9,72 @@ import Navbar from "react-bootstrap/Navbar";
+          className={styles.unbutton}
+        >
+          {" "}
+          🌟 {caseStudy.title}
```
The color of the star kinda clashes with the background imo, I would do no bullet point and make the text not italic, unless the star has signficance.

hschafer commented:
> Here are some thing i noticed/think should change.
> 
> 1. No next chapter on the home page if possible.
> 2. The text is really hard to read with that shade of purple, I'd either make it a darker purple with lighter text, or a lighter background with black text.
> 3. The BM tag is aligned to the center of the page not the column, idk why this is.

Fixed all of these! The BM tag had an issue with CSS specificity that I introduced in the CSS update 😭

hschafer commented:
Whoops, forgot to actually push the commits :P
